app/exec/extension/_lib/vsix-manifest-builder.ts

```diff
@@ -396,6 +396,39 @@ export class VsixManifestBuilder extends ManifestBuilder {
 				}
 				this.addProperty("Microsoft.VisualStudio.Services.Content.Pricing.PriceCalculator", value.toString());
 				break;
+			case "galleryproperties":
+				/**
+				 * The Gallery Properties would be a generic array of JSON elements
+				 */
+				if (_.isArray(value)) {
+					value.forEach((propertyGroup: any) => {						
+						Object.keys(propertyGroup).forEach((propertyKey: string) => {
+							if(propertyKey && propertyGroup[propertyKey]) {
+
+								// Property ID would be in upper camel case (First letter Capital)
+								let ucck: string = propertyKey.charAt(0).toUpperCase() + propertyKey.slice(1);
+								trace.debug("Property key %s.", ucck);
+								trace.debug("Property value %s.", String(propertyGroup[propertyKey]));
+								
+								let propertyName: string = "Microsoft.VisualStudio.Services.GalleryProperties." + ucck;
+
+								// Check for duplicates								
+								let existingProperties = _.get<any[]>(this.data, "PackageManifest.Metadata[0].Properties[0].Property", []);
```

Comment:
`string[]` instead of `any[]`?

Reply:
I have copied this code from the addProperty() where we are using any[]. IMO let it be same everywhere

Reply:
Ok